### DISCUSSION TITLE
Question about ESlint

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#64300] [Help] [answered]
### Summary

How do I turn ESLint on and off?

Apparently I have a website were the ESLint is ON, and it gives me some warnings on a code structure.

Another site seems like ESLint is OFF, and it doesn't give me any warnings on a code structure.

I think at one point I turn ESLint off on one site because I was using a Google Ad code that was not working properly and giving me errors.....  it was a problem from Google that likely they never fixed..

**Comment by icyJoseph** [comment]
ESLint is a development time tool, which needs to configured, in order to give you warnings and errors. These warnings, and errors, are coming from running various rules against the code you write. 

Whenever ESLint finds errors, it exits with a non zero code, which can be used, to, for example, fail a build step, so that the developer has to fix the ESLint errors, in order to be able to build the application. Some include these in the `git` commands as well, so that a developer can't commit code that has linting issues. Others make it so that, warnings exit with a non zero code, triggering the afore mentioned behavior.

Without looking at the repositories it is hard to tell what ESLint configurations you might have, or not. Note that ESLint errors, are not the same as TypeScript errors, necessarily, but there might be some overlap, just like there's also overlap between some ESLint rules, and some Prettier formatting rules.

**Reply by Elindo586** [reply]
This would be the repo that I "think" doesn't have warnings from ESLint...

https://github.com/Elindo586/nextprojectb

^^  On a different bag of cookies.  I am currently fixing some issues from Google Lighthouse and it has TONS of performance issues, along with SOME issues on Best Practices.... Accessibility, SEO and PWA are good on green.

**Reply by icyJoseph** [reply]
There's an `.eslintrc.json` file, it should work. Run `npx next lint` in your terminal, or `npm run lint`

**Reply by Elindo586** [reply]
How do I know if it is working?

```
PS C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb> npx next lint
Referenced from: C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb\.eslintrc.json
PS C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb> npm run lint

> nextprojectb@0.1.0 lint
> next lint

Failed to load config "next/core-web-vitals" to extend from.
Referenced from: C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb\.eslintrc.json
```

**Reply by icyJoseph** [reply]
Aha, something is wrong in that repository, let me take a look again...

**Reply by icyJoseph** [reply, marked as answer]
Oh I see, you have eslint-config-next, at 0.2.4, that's very old. See here: 

- https://github.com/Elindo586/nextprojectb/blob/main/package.json

```json
  "devDependencies": {
    "eslint": "8.15.0",
    "eslint-config-next": "^0.2.4"
  }
```

Your `eslint-config-next`, should be at the same version as your `next` package:

```json
    "gsap": "^3.11.0",
    "next": "^13.1.1",
    "nodemailer": "^6.7.5",
    "openai": "^4.12.4",
```

So, `npm install --dev eslint-config-next@13.1.1`, should do the trick.

**Reply by Elindo586** [reply]
Well... it worked... now I got to fix a bunch of errors before I can upload again
